src/Tmds.LinuxAsync/IOUringAsyncEngine.IOUringExecutionQueue.cs

```diff
@@ -232,7 +247,7 @@ public unsafe void SubmitAndWait(bool mayWait)
                 bool waitForCompletion = !operationsRemaining && mayWait;
 
                 // io_uring_enter
-                SubmitResult result = _ring!.SubmitAndWait(minComplete: waitForCompletion ? 1U : 0, out _);
+                var result = _ring!.SubmitAndWait(minComplete: waitForCompletion ? 1U : 0, out _);
```

Comment:
Currently we never wait for a completions when there are submissions remaining.
So if the eventfd read was not yet submitted, we can never be waiting.

Is it possible we now wait for completions without being sure the eventfd read is submitted?
That would mean we may not able to unblock the wait by writing to the eventfd.

Reply:
I think that could happen, yes. I have to put some more thought into it. Off the top of my hread we could:
* somehow track the submission of the eventfd read?
* only add the eventfd read as (likely) only SQE if there are no other SQEs prepared by the time of calling SubmitAndWait.